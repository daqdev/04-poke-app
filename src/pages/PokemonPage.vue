<template>
  <h1>Quien es este Pokemon</h1>
  <pokemon-image/>
  <pokemon-options/>

  <!-- opciones -->
</template>

<script>
import PokemonImage from '@/components/PokemonImage.vue'
import PokemonOptions from '@/components/PokemonOptions.vue'
export default {
  components: { PokemonImage, PokemonOptions },

}
</script>



