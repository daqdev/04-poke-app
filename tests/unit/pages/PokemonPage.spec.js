import { mount, shallowMount } from "@vue/test-utils"
import PokemonPage from '@/pages/PokemonPage'
import { pokemonzMok } from "../mocks/pokemons.mock"



describe('PokemonPage Component', () => { 

let wrapper
beforeEach(()=>{
    wrapper=shallowMount(PokemonPage)
})

    test('debe hacer match con el snapshot',()=>{
        expect(wrapper.html()).toMatchSnapshot()
    })

    test ('debe llamar al pokemon mix array en el mount',()=>{
        //aca tenemos q poner el spy antes de iniciar la pagina
        //ya que el metodo que queremos escuchar es parte de la inicializacion
        const spyPage = jest.spyOn(PokemonPage.methods, 'mixPokemonArray')
        wrapper = shallowMount(PokemonPage)
        expect(spyPage).toHaveBeenCalled()
    })

    test ('debe hacer match con  el snapshot cuando carga los pokemon', ()=>{

        const wrapper = shallowMount(PokemonPage, {
        // const wrapper = mount(PokemonPage, {
            data(){
                return {
                    pokemonArr: pokemonzMok,
                    pokemon : pokemonzMok[0],
                    showPokemon:false,
                    respondio:false,
                    resultado:'Todavia no respondio!'
                }
            }
        })
        expect(wrapper.html()).toMatchSnapshot()


    })
 })