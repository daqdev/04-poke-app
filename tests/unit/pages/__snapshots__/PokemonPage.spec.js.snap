// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage Component debe hacer match con  el snapshot cuando carga los pokemon 1`] = `
<div>
  <h1>Quien es este Pokemon???</h1>
  <!--v-if-->
  <pokemon-image-stub pokemonid="1" showpokemon="false"></pokemon-image-stub>
  <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub><button>Nuevo Juego</button>
</div>
`;

exports[`PokemonPage Component debe hacer match con el snapshot 1`] = `<h1>Cargando ...</h1>`;
